pkg/parser/parser_test.go

```diff
@@ -1203,6 +1203,7 @@ func TestDBAStmt(t *testing.T) {
 		{`SHOW PROFILE CPU FOR QUERY 2 LIMIT 1,1`, true, "SHOW PROFILE CPU FOR QUERY 2 LIMIT 1,1"},
 		{`SHOW PROFILE CPU, MEMORY, BLOCK IO, CONTEXT SWITCHES, PAGE FAULTS, IPC, SWAPS, SOURCE FOR QUERY 1 limit 100`, true, "SHOW PROFILE CPU, MEMORY, BLOCK IO, CONTEXT SWITCHES, PAGE FAULTS, IPC, SWAPS, SOURCE FOR QUERY 1 LIMIT 100"},
 		{`SHOW MASTER STATUS`, true, "SHOW MASTER STATUS"},
+		{`SHOW BINARY LOG STATUS`, true, "SHOW MASTER STATUS"},
```

Comment:
I slightly prefer we restore the two SQL to SHOW BINARY LOG STATUS, because SHOW MASTER STATUS is deprecated.

Reply:
This is possible, but needs more changes.

From `git diff -- . ':(exclude)pkg/parser/parser.go'`

```diff
diff --git a/pkg/executor/builder.go b/pkg/executor/builder.go
index d2a9e98e9a..a3761be1a3 100644
--- a/pkg/executor/builder.go
+++ b/pkg/executor/builder.go
@@ -875,7 +875,7 @@ func (b *executorBuilder) buildShow(v *plannercore.PhysicalShow) exec.Executor {
 		Extractor:             v.Extractor,
 		ImportJobID:           v.ImportJobID,
 	}
-	if e.Tp == ast.ShowMasterStatus {
+	if e.Tp == ast.ShowMasterStatus || e.Tp == ast.ShowBinlogStatus {
 		// show master status need start ts.
 		if _, err := e.Ctx().Txn(true); err != nil {
 			b.err = err
diff --git a/pkg/executor/show.go b/pkg/executor/show.go
index f4b9847a81..0ac276bbea 100644
--- a/pkg/executor/show.go
+++ b/pkg/executor/show.go
@@ -253,7 +253,7 @@ func (e *ShowExec) fetchAll(ctx context.Context) error {
 		return e.fetchShowPlugins()
 	case ast.ShowProfiles:
 		// empty result
-	case ast.ShowMasterStatus:
+	case ast.ShowMasterStatus, ast.ShowBinlogStatus:
 		return e.fetchShowMasterStatus()
 	case ast.ShowPrivileges:
 		return e.fetchShowPrivileges()
diff --git a/pkg/parser/ast/dml.go b/pkg/parser/ast/dml.go
index 7636ebf98d..69ea4efb3d 100644
--- a/pkg/parser/ast/dml.go
+++ b/pkg/parser/ast/dml.go
@@ -2978,6 +2978,7 @@ const (
 	ShowCreateResourceGroup
 	ShowImportJobs
 	ShowCreateProcedure
+	ShowBinlogStatus
 )
 
 const (
@@ -3067,6 +3068,8 @@ func (n *ShowStmt) Restore(ctx *format.RestoreCtx) error {
 
 	ctx.WriteKeyWord("SHOW ")
 	switch n.Tp {
+	case ShowBinlogStatus:
+		ctx.WriteKeyWord("BINARY LOG STATUS")
 	case ShowCreateTable:
 		ctx.WriteKeyWord("CREATE TABLE ")
 		if err := n.Table.Restore(ctx); err != nil {
diff --git a/pkg/parser/parser.y b/pkg/parser/parser.y
index e13d983883..1aac154278 100644
--- a/pkg/parser/parser.y
+++ b/pkg/parser/parser.y
@@ -11362,7 +11362,7 @@ ShowStmt:
 |	"SHOW" "BINARY" "LOG" "STATUS"
 	{
 		$$ = &ast.ShowStmt{
-			Tp: ast.ShowMasterStatus,
+			Tp: ast.ShowBinlogStatus,
 		}
 	}
 |	"SHOW" OptFull "PROCESSLIST"
diff --git a/pkg/parser/parser_test.go b/pkg/parser/parser_test.go
index 9eec7ee1df..17bfc31a78 100644
--- a/pkg/parser/parser_test.go
+++ b/pkg/parser/parser_test.go
@@ -1203,7 +1203,7 @@ func TestDBAStmt(t *testing.T) {
 		{`SHOW PROFILE CPU FOR QUERY 2 LIMIT 1,1`, true, "SHOW PROFILE CPU FOR QUERY 2 LIMIT 1,1"},
 		{`SHOW PROFILE CPU, MEMORY, BLOCK IO, CONTEXT SWITCHES, PAGE FAULTS, IPC, SWAPS, SOURCE FOR QUERY 1 limit 100`, true, "SHOW PROFILE CPU, MEMORY, BLOCK IO, CONTEXT SWITCHES, PAGE FAULTS, IPC, SWAPS, SOURCE FOR QUERY 1 LIMIT 100"},
 		{`SHOW MASTER STATUS`, true, "SHOW MASTER STATUS"},
-		{`SHOW BINARY LOG STATUS`, true, "SHOW MASTER STATUS"},
+		{`SHOW BINARY LOG STATUS`, true, "SHOW BINARY LOG STATUS"},
 		{`SHOW PRIVILEGES`, true, "SHOW PRIVILEGES"},
 		// for show character set
 		{"show character set;", true, "SHOW CHARSET"},
diff --git a/pkg/planner/core/planbuilder.go b/pkg/planner/core/planbuilder.go
index 9f5adbca51..8643d8649e 100644
--- a/pkg/planner/core/planbuilder.go
+++ b/pkg/planner/core/planbuilder.go
@@ -5363,6 +5363,9 @@ func buildShowSchema(s *ast.ShowStmt, isView bool, isSequence bool) (schema *exp
 	var names []string
 	var ftypes []byte
 	switch s.Tp {
+	case ast.ShowBinlogStatus:
+		names = []string{"File", "Position", "Binlog_Do_DB", "Binlog_Ignore_DB", "Executed_Gtid_Set"}
+		ftypes = []byte{mysql.TypeVarchar, mysql.TypeLonglong, mysql.TypeVarchar, mysql.TypeVarchar, mysql.TypeVarchar}
 	case ast.ShowProcedureStatus, ast.ShowFunctionStatus:
 		return buildShowProcedureSchema()
 	case ast.ShowTriggers:
diff --git a/pkg/planner/core/planbuilder_test.go b/pkg/planner/core/planbuilder_test.go
index 81ee1e7bac..365fcccf5d 100644
--- a/pkg/planner/core/planbuilder_test.go
+++ b/pkg/planner/core/planbuilder_test.go
@@ -49,6 +49,7 @@ type visit struct {
 func TestShow(t *testing.T) {
 	node := &ast.ShowStmt{}
 	tps := []ast.ShowStmtType{
+		ast.ShowBinlogStatus,
 		ast.ShowEngines,
 		ast.ShowDatabases,
 		ast.ShowTables,
```

@lance6716  Should I push this?

Reply:
Another option would be to restore both `SHOW MASTER STATUS` and `SHOW BINARY LOG STATUS` as the new `SHOW BINARY LOG STATUS`, but that might lead to compatibility issues.

Reply:
Yes, I prefer you push it.

Reply:
Pushed as 01ff455beb256f7f5d9f6965bf9bc2d6b1f2a127